src/Fixer/PhpUnit/PhpUnitMethodCasingFixer.php

```diff
@@ -0,0 +1,277 @@
+     * {@inheritdoc}
+     */
+    public function getDefinition()
+    {
+        return new FixerDefinition(
+            'Enforce camel (or snake) case for PHPUnit test methods, following configuration.',
+            [
+                new CodeSample(
+                    '<?php
+class MyTest extends \\PhpUnit\\FrameWork\\TestCase
+{
+    public function test_my_code() {}
+}
+'
+                ),
+                new CodeSample(
+                    '<?php
+class MyTest extends \\PhpUnit\\FrameWork\\TestCase
+{
+    public function testMyCode() {}
+}
+',
+                    ['case' => self::SNAKE_CASE]
+                ),
+            ]
+        );
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function isCandidate(Tokens $tokens)
+    {
+        return $tokens->isAllTokenKindsFound([T_CLASS, T_FUNCTION]);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function applyFix(\SplFileInfo $file, Tokens $tokens)
+    {
+        $phpUnitTestCaseIndicator = new PhpUnitTestCaseIndicator();
+        foreach ($phpUnitTestCaseIndicator->findPhpUnitClasses($tokens) as $indexes) {
+            $this->applyCasing($tokens, $indexes[0], $indexes[1]);
+        }
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function createConfigurationDefinition()
+    {
+        return new FixerConfigurationResolver([
+            (new FixerOptionBuilder('case', 'Apply camel or snake case to test methods'))
+                ->setAllowedValues([self::CAMEL_CASE, self::SNAKE_CASE])
+                ->setDefault(self::CAMEL_CASE)
+                ->getOption(),
+        ]);
+    }
+
+    /**
+     * @param Tokens $tokens
+     * @param int    $startIndex
+     * @param int    $endIndex
+     */
+    private function applyCasing(Tokens $tokens, $startIndex, $endIndex)
+    {
+        for ($index = $endIndex - 1; $index > $startIndex; --$index) {
+            if (!$this->isTestMethod($tokens, $index)) {
+                continue;
+            }
+
+            $functionNameIndex = $tokens->getNextMeaningfulToken($index);
+            $functionName = $tokens[$functionNameIndex]->getContent();
+            $newFunctionName = $this->updateMethodCasing($functionName);
+
+            if ($newFunctionName !== $functionName) {
+                $tokens[$functionNameIndex] = new Token([T_STRING, $newFunctionName]);
+            }
+
+            $docBlockIndex = $this->getDocBlockIndex($tokens, $index);
+            if ($this->hasDocBlock($tokens, $index)) {
+                $this->updateDocBlock($tokens, $docBlockIndex);
+            }
+        }
+    }
+
+    /**
+     * @param string $functionName
+     *
+     * @return string
+     */
+    private function updateMethodCasing($functionName)
+    {
+        if (self::CAMEL_CASE === $this->configuration['case']) {
+            $newFunctionName = $functionName;
+            $newFunctionName = str_replace('_', ' ', $newFunctionName);
+            $newFunctionName = ucwords($newFunctionName);
```

Comment:
`ucwords` takes delimiter as 2nd param, so you can put there `_` and avoid remove `str_replace` above.